app/models/miq_ae_yaml_import.rb

```diff
@@ -222,13 +222,74 @@ def process_method(class_obj, ruby_method_file_name, method_yaml)
     if method_attributes['location'] == 'inline'
       data = load_method_ruby(ruby_method_file_name)
       method_yaml.store_path('object', 'attributes', 'data', data) if data
+    elsif method_attributes['location'] == 'playbook'
+      convert_playbook_attributes(method_attributes['options'], ruby_method_file_name)
     end
     method_obj = MiqAeMethod.find_by(:name => method_attributes['name'], :class_id => class_obj.id) unless class_obj.nil?
     track_stats('method', method_obj)
     method_obj ||= add_method(class_obj, method_yaml) unless @preview
     method_obj
   end
 
+  def convert_playbook_attributes(options, ruby_method_file_name)
+    invalid_attributes = []
+    ae_manager = ManageIQ::Providers::EmbeddedAnsible::AutomationManager
+
+    %w[repository playbook credential vault_credential cloud_credential].each do |attr|
+      next unless options["#{attr}_name".to_sym]
+      klass = case attr
+              when 'repository'
+                AnsibleRepositoryController.model
+              when 'playbook'
+                ae_manager::Playbook
+              when 'credential'
+                ae_manager::Credential
+              when 'vault_credential'
+                ae_manager::VaultCredential
+              when 'cloud_credential'
+                ae_manager::CloudCredential
+              end
+      convert_playbook_attribute(options, klass, attr, invalid_attributes)
+    end
+
+    unless invalid_attributes.empty?
+      error_msg = "Error: Playbook method '#{ruby_method_file_name}' contains some invalid attributes:"
+      invalid_attributes.each do |attr|
+        error_msg += "<br> * #{playbook_attr_error_msg(attr, options)}"
+      end
+      raise MiqAeException::AttributeNotFound, error_msg
+    end
+    options.except!(:repository_name, :playbook_name, :credential_name, :vault_credential_name, :cloud_credential_name)
+  end
+
+  def convert_playbook_attribute(options, klass, attr, invalid_attributes)
+    related_obj = klass.find_by(:name => options["#{attr}_name".to_sym])
+
+    if related_obj
+      options["#{attr}_id".to_sym] = related_obj.id.to_s
```

Comment:
Any reason for ```to_s```?

Reply:
@lfu If you create a new Playbook Method via UI, these IDs are stored as a string values. Also when I tried to change them to numbers instead of strings, at least 1 page stopped working. Editing page did not load these values, so I think that UI assume strings there.